appintro/src/main/java/com/github/paolorotolo/appintro/AppIntroBase.kt

```diff
@@ -648,6 +705,55 @@ abstract class AppIntroBase : AppCompatActivity(), AppIntroViewPager.OnNextPageR
         }
     }
 
+    private fun onDenyPressed(permissionResults: HashMap<String, Int>) {
+        for ((permName) in permissionResults) {
+
+            // Permission is denied for the first time (never ask again box is not checked).
+            // Ask again explaining the usage of the permission (Show an AlertDialog or Snackbar)
+            // shouldShowRequestPermissionRationale will return true
+
+            if (ActivityCompat.shouldShowRequestPermissionRationale(this, permName)) {
+                if (!isPermissionRequired) {
+                    // If the permission request was a success, we remove
+                    // the permission from the permissionsArray,
+                    // and also remove the permission slide from permissionSlides.
+                    permissionsArray.remove(requestedPermission)
+                    permissionSlides.remove(pager.getCurrentSlideNumber(fragments.size))
+                    moveForward()
+                }
+                onUserDeniedPermission(permName)
+            } else {
+                // Permission is disabled (never ask again is checked)
+                // shouldShowRequestPermissionRationale will return false.
+                // Ask the user to go to settings to enable permission.
+                onUserDisabledPermission(permName)
+                break
+            }
+        }
+    }
+    private fun moveForward() {
+        // Check if next slide is the last one, then proceed.
+        if (pager.currentItem + 1 == slidesNumber) {
+            goToNextSlide(true)
+        } else {
+            goToNextSlide(false)
+        }
+    }
+
```

Comment:
This whole function is probably not needed. You can set a default parameter for `goToNextSlide`:

```kotlin
protected fun goToNextSlide(isLastSlide: Boolean = pager.currentItem == pagerAdapter.count - 1) {
```